domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/TimelineEvent.kt

```diff
@@ -42,25 +42,33 @@ data class TimelineEvent(
    * The date and time when the event occurred.
    */
   val timestamp: Instant,
+  /**
+   * An optional correlationId for this and any other TimelineEvents that occurred at the same time (i.e. within same
+   * atomic action). For example, this could be an update to a Goal and one or more of its child Steps.
+   */
+  val correlationId: UUID?,
 ) {
   companion object {
     fun newTimelineEvent(
+      reference: UUID = UUID.randomUUID(),
       sourceReference: String,
       eventType: TimelineEventType,
       contextualInfo: String? = null,
       prisonId: String,
       actionedBy: String,
       actionedByDisplayName: String? = null,
       timestamp: Instant = Instant.now(),
+      correlationId: UUID = UUID.randomUUID(),
```

Comment:
Re: my comment above, I can 100% see the use case where correlationId needs to be passed in / specified by the caller, but not sure about reference